geozero/src/wkb/wkb_reader.rs

```diff
@@ -130,16 +128,13 @@ pub(crate) fn read_wkb_header<R: Read>(raw: &mut R) -> Result<WkbInfo> {
     let is_little_endian = byte_order != 0;
     let endian = Endian::from(is_little_endian);
     let type_id = raw.ioread_with::<u32>(endian)?;
-    let base_type = WKBGeometryType::from_u32(type_id % 1000);
     let type_id_dim = type_id / 1000;
-    let has_z = type_id_dim == 1 || type_id_dim == 3;
-    let has_m = type_id_dim == 2 || type_id_dim == 3;
 
     let info = WkbInfo {
         endian,
-        base_type,
-        has_z,
-        has_m,
+        base_type: WKBGeometryType::from_u32(type_id % 1000),
+        has_z: matches!(type_id_dim, 1 | 3),
```

Comment:
i'm actually following your pattern here - you are right, matches! is easier to read here